### ISSUE TITLE
Amazon crash when requesting purchase

### ISSUE THREAD

**Issue by mandrewsan** [dooboolab/react-native-iap#1869]
**Description**

App crashes when calling `RNIap.requestPurchase({ sku })` from an Amazon Kindle Fire.

Main error is saying: `RNIapAmazonModule.buyItemByType got 10 arguments, expected 3`

**Expected Behavior**

This was previously working, prior to upgrading.

**Environment:**

- react-native-iap: 9.0.3
- react-native: 0.69.2
- Platforms (iOS, Android, emulator, simulator, device): Amazon Kindle, real device

**To Reproduce**

My code looks like this:

```
import * as RNIap from "react-native-iap";
...
async componentDidMount() {
    ...
    try {
      await RNIap.initConnection();
      const productsList = await RNIap.getProducts(productIds);
      this.setState({ productsList });
    } catch (err) {
      ...
    }
    this.addPurchaseListeners();
}

requestPurchase = async (sku) => {
    try {
      await RNIap.requestPurchase({ sku });
    } catch (err) {
      Alert.alert("Purchase Error", err.message);
    }
  };
```

---

[Optional] **Additional Context: Resulting Error**

```
2022-08-17 12:44:40.930 11549-11624/PACKAGE_NAME E/unknown:ReactNative: Exception in native call
    com.facebook.react.bridge.NativeArgumentsParseException: RNIapAmazonModule.buyItemByType got 10 arguments, expected 3
        at com.facebook.react.bridge.JavaMethodWrapper.invoke(JavaMethodWrapper.java:349)
        at com.facebook.react.bridge.JavaModuleWrapper.invoke(JavaModuleWrapper.java:188)
        at com.facebook.react.bridge.queue.NativeRunnable.run(Native Method)
        at android.os.Handler.handleCallback(Handler.java:873)
        at android.os.Handler.dispatchMessage(Handler.java:99)
        at com.facebook.react.bridge.queue.MessageQueueThreadHandler.dispatchMessage(MessageQueueThreadHandler.java:27)
        at android.os.Looper.loop(Looper.java:211)
        at com.facebook.react.bridge.queue.MessageQueueThreadImpl$4.run(MessageQueueThreadImpl.java:228)
        at java.lang.Thread.run(Thread.java:764)
```

**Comment by andresesfm** [comment]
Thank you for the report @mandrewsan . Fix is here: https://github.com/dooboolab/react-native-iap/pull/1870

**Comment by andresesfm** [comment]
Fix released on `9.0.4`

**Comment by mandrewsan** [comment]
Thank you! The fix worked.
